homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,383 @@
+        key="vapor_pressure",
+        translation_key="vapor_pressure",
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        imperial_suggested_unit=UnitOfPressure.INHG,
+        suggested_display_precision=5,
+    ),
+    ## Wind Sensors
+    WeatherFlowWindSensorEntityDescription(
+        key="wind_gust",
+        translation_key="wind_gust",
+        device_class=SensorDeviceClass.WIND_SPEED,
+    ),
+    WeatherFlowWindSensorEntityDescription(
+        key="wind_lull",
+        translation_key="wind_lull",
+        device_class=SensorDeviceClass.WIND_SPEED,
+    ),
+    WeatherFlowWindSensorEntityDescription(
+        key="wind_speed",
+        device_class=SensorDeviceClass.WIND_SPEED,
+        event_subscriptions=[EVENT_RAPID_WIND, EVENT_OBSERVATION],
+    ),
+    WeatherFlowWindSensorEntityDescription(
+        key="wind_speed_average",
+        translation_key="wind_speed_average",
+        device_class=SensorDeviceClass.WIND_SPEED,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction",
+        translation_key="wind_direction",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+        event_subscriptions=[EVENT_RAPID_WIND, EVENT_OBSERVATION],
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="wind_direction_average",
+        translation_key="wind_direction_average",
+        icon="mdi:compass-outline",
+        native_unit_of_measurement=DEGREE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WeatherFlow sensors using config entry."""
+
+    @callback
+    def async_add_sensor(device: WeatherFlowDevice) -> None:
+        """Add WeatherFlow sensor."""
+        LOGGER.debug("Adding sensors for %s", device)
+
+        sensors: list[WeatherFlowSensorEntity | WeatherFlowAirDensitySensorEntity] = [
+            WeatherFlowSensorEntity(
+                device=device,
+                description=description,
+                is_metric=(hass.config.units == METRIC_SYSTEM),
+            )
+            for description in SENSORS
+            if hasattr(device, description.key)
+        ]
+
+        sensors = sensors + [
+            WeatherFlowAirDensitySensorEntity(
+                device=device,
+                description=description,
+                is_metric=(hass.config.units == METRIC_SYSTEM),
+            )
+            for description in VOC_SENSORS
+            if hasattr(device, description.key)
+        ]
+
+        async_add_entities(sensors)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_{SENSOR_DOMAIN}",
```

Comment:
```suggestion
            f"{DOMAIN}_{config_entry.entry_id}_add",
```

Reply:
Would be better to have a function that formatted the signal name based on the entry_id so you can use it in both places